google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/AuthorizedView.java

```diff
@@ -0,0 +1,309 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigtable.admin.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.admin.v2.AuthorizedViewName;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ByteString;
+import java.util.Map.Entry;
+import javax.annotation.Nonnull;
+
+/**
+ * A class that wraps the {@link com.google.bigtable.admin.v2.AuthorizedView} protocol buffer
+ * object.
+ *
+ * <p>An AuthorizedView represents subsets of a particular table based on rules. The access to each
+ * AuthorizedView can be configured separately from the Table.
+ *
+ * <p>Users can perform read/write operation on an AuthorizedView by providing an authorizedView id
+ * besides a table id, in which case the semantics remain identical as reading/writing on a Table
+ * except that visibility is restricted to the subset of the Table that the AuthorizedView
+ * represents.
+ */
+public final class AuthorizedView {
+  private final com.google.bigtable.admin.v2.AuthorizedView proto;
+
+  /**
+   * Wraps the protobuf. This method is considered an internal implementation detail and not meant
+   * to be used by applications.
+   */
+  @InternalApi
+  public static AuthorizedView fromProto(
+      @Nonnull com.google.bigtable.admin.v2.AuthorizedView proto) {
+    return new AuthorizedView(proto);
+  }
+
+  private AuthorizedView(@Nonnull com.google.bigtable.admin.v2.AuthorizedView proto) {
+    Preconditions.checkNotNull(proto);
+    Preconditions.checkArgument(!proto.getName().isEmpty(), "AuthorizedView must have a name");
+    Preconditions.checkArgument(
+        proto.hasSubsetView(), "AuthorizedView must have a subset_view field");
+    this.proto = proto;
+  }
+
+  /** Gets the authorized view's id. */
+  public String getId() {
+    // Constructor ensures that name is not null.
+    AuthorizedViewName fullName = AuthorizedViewName.parse(proto.getName());
+
+    //noinspection ConstantConditions
+    return fullName.getAuthorizedView();
+  }
+
+  /** Gets the id of the table that owns this authorized view. */
+  public String getTableId() {
+    // Constructor ensures that name is not null.
+    AuthorizedViewName fullName = AuthorizedViewName.parse(proto.getName());
+
+    //noinspection ConstantConditions
+    return fullName.getTable();
+  }
+
+  /** Returns whether this authorized view is deletion protected. */
+  public boolean isDeletionProtected() {
+    return proto.getDeletionProtection();
+  }
+
+  /** Gets the implementation of this authorized view, which currently can only be a subset view. */
+  public AuthorizedViewImpl getAuthorizedViewImpl() {
```

Comment:
is this only used for testing? Please annotate with `@VisibleForTesting`

Reply:
I think we do want this to be public so that users can inspect what's in their authorized views.